### DISCUSSION TITLE
Video Playback Overlay

### DISCUSSION THREAD

**Discussion by jacob-willden** [PseudoTV/PseudoTV_Live#38] [Q&A]
Thanks for creating and maintaining this plugin! I have a question about overlaying images over each channel, like was mentioned in an [old thread on the Kodi forums](https://forum.kodi.tv/showthread.php?tid=316537). 

I'm creating a Kodi add-on that provides image overlays during videos, and I've tried creating the overlay using `xbmcgui.WindowDialog`, but whenever the window is shown (using the `show()` method), it prevents user access to the rest of the video interface. After I heard about this plugin, I dived into its code (particularly the overlay.py file), and was surprised to find out that the overlays also appear to be created using `WindowDialog`.

My question is: how does this plugin display the image overlays while still allowing the user to control video playback? Thank you so much for your time and help!

**Comment by Lunatixz** [comment]
It doesn't, unfortunately the `WindowDialog` class gains complete control of all actions and clicks.

Looking at your post. If your overlay is not for video, you can use `WindowXML` which does not gain action control.

**Comment by jacob-willden** [comment]
Interesting, I'll try that out. Thanks so much for the help!

**Reply by Lunatixz** [reply]
You can find docs here: https://xbmc.github.io/docs.kodi.tv/master/kodi-base/dd/d4d/group__python__xbmcgui__window__xml.html

**Comment by jacob-willden** [comment]
By the way, I found another way to accomplish this on a [Kodi Forums thread](https://forum.kodi.tv/showthread.php?tid=217212), which actually uses `object`. Once I get the functionality fully working, I'll share it here for the benefit of future developers. Thanks again!

**Reply by Lunatixz** [reply]
I believe it's technically impossible! The dialog class intercepts all actions while the window is open. Keep me updated, THX

**Comment by jacob-willden** [comment]
I've got it fully working now. This is the class for the overlay ([source](https://forum.kodi.tv/member.php?action=profile&uid=143840)):
```
class OverlayScreen(object):
    def __init__(self):
        self.showing = False
        self.window = xbmcgui.Window(12005) # Inheriting from 12005 (fullscreenvideo) puts the overlay in front of the video, but behind the video interface
        origin_x = 0
        origin_y = 0
        window_w = self.window.getWidth()
        window_h = self.window.getHeight()
        self._background = xbmcgui.ControlImage(origin_x, origin_y, window_w, window_h, os.path.join(addonpath,"resources","skins","default","media","overlay.png"))

    def show(self):
        self.showing=True
        self.window.addControl(self._background)

    def hide(self):
        self.showing=False
        self.window.removeControl(self._background)

    def _close(self):
        if self.showing:
            self.hide()
        else:
            pass
        try:
            self.window.clearProperties()
            print("OverlayText window closed")
        except: pass
```
Once inside the main execution section of the script (`if __name__ == '__main__':`) and once `xbmc.Player().isPlaying()` is true, a variable can be assigned `OverlayScreen()` and then its methods can be called. Thanks for sending me in the right direction for this!

**Reply by Lunatixz** [reply]
IC, you're adding a new control to an existing window... Interesting approach though it has limitations.  THX for letting me know.
